proxy/coco/src/state.rs

```diff
@@ -97,8 +76,12 @@ impl State {
     /// # Panics
     ///
     ///   * Unable to acquire the lock.
-    pub fn has_commit(&self, urn: &RadUrn, oid: impl Into<git2::Oid>) -> Result<bool, Error> {
-        Ok(self.api.storage().has_commit(urn, oid.into())?)
+    pub async fn has_commit(&self, urn: RadUrn, oid: impl Into<git2::Oid>) -> Result<bool, Error> {
+        let oid = oid.into();
+        Ok(self
+            .api
+            .with_storage(move |storage| storage.has_commit(&urn, oid))
+            .await??)
```

Comment:
Before you ask: `??` is pronounced `The wat? operator`, was stabilised in Rust'98, and is part of the Rust Dawallah Level II certification programme.

Reply:
I'm looking forward to the interrobang RFC. `?!` will be the `The WAT?! operator` and will indicate to code reviewers that they should move along